### PR TITLE
Add: gitleaks sast tools

### PR DESCRIPTION
**Pull request by dangdong163**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: gitleaks.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package:
+  name: gitleaks
+  version: 8.18.2
+  epoch: 0
+  description: SAST tool for detecting and preventing hardcoded secrets like passwords, api keys, and tokens in git repos
+  target-architecture:
+    - x86_64
+    - aarch64a
```
Comment (ajayk) [inline gitleaks.yaml:8]:
```suggestion
```

Comment (dangdong163) [reply gitleaks.yaml]: i updated in new commit

File: gitleaks.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
.../gitleaks/go-gitdiff@v0.9.0 github.com/h2non/filetype@v1.1.3 github.com/rs/zerolog@v1.26.1 github.com/spf13/cobra@v1.2.1 github.com/spf13/viper@v1.8.1 github.com/stretchr/testify@v1.7.0 github.com/aymanbagabas/go-osc52/v2@v2.0.1 github.com/lucasb-eyer/go-colorful@v1.2.0 github.com/mattn/go-isatty@v0.0.17 github.com/mattn/go-runewidth@v0.0.14 github.com/muesli/reflow@v0.2.1-0.20210115123740-9e1d0d53df68 github.com/muesli/termenv@v0.15.1 github.com/rivo/uniseg@v0.2.0 github.com/davecgh/go-spew@v1.1.1 github.com/fsnotify/fsnotify@v1.4.9 github.com/hashicorp/hcl@v1.0.0 github.com/inconshreveable/mousetrap@v1.0.0 github.com/lucasjones/reggen@v0.0.0-20200904144131-37ba4fa293bb github.com/magiconair/properties@v1.8.5 github.com/mitchellh/mapstructure@v1.4.1 github.com/pelletier/go-toml@v1.9.3 github.com/pmezard/go-difflib@v1.0.0 github.com/spf13/afero@v1.6.0 github.com/spf13/cast@v1.3.1 github.com/spf13/jwalterweatherman@v1.1.0 github.com/spf13/pflag@v1.0.5 github.com/subosito/gotenv@v1.2.0 golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c golang.org/x/sys@v0.6.0 golang.org/x/text@v0.3.8 gopkg.in/ini.v1@v1.62.0 gopkg.in/yaml.v2@v2.4.0 gopkg.in/yaml.v3@v3.0.0-20210107192922-496545a6307b
```
Comment (tuananh) [inline gitleaks.yaml:27]: uncomment this maybe? :)

Comment (dangdong163) [reply gitleaks.yaml]: sure, i updated in new commit. i have tested on apko also and add test steps :#

File: gitleaks.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/gitleaks/gitleaks
+      expected-commit: ac4b5146b0f112df989b4374abb2b12799e37cba
+      tag: v${{package.version}}
+
+  - runs: |
+      make build
+      mv gitleaks /usr/local/bin/
```
Comment (ajayk) [inline gitleaks.yaml:27]:
```suggestion
      mkdir -p ${{targets.destdir}}/usr/bin
      mv gitleaks ${{targets.destdir}}/usr/bin/gitleaks
```

Comment (dangdong163) [reply gitleaks.yaml]: i updated path run the command

**Review by ajayk — APPROVED** [review]

**Pull request merged**